File: sdk/typescript/src/memory.ts
import { MemoryClient, MemoryConfig } from "./client";

export class Memory {
    private client: MemoryClient;

    constructor(config: MemoryConfig) {
        this.client = new MemoryClient(config);
    }

    async store(content: string, metadata?: Record<string, any>) {
        return this.client.request("POST", "/memory/store", { content, metadata });
    }

    async recall(query: string, limit: number = 10) {
        return this.client.request("POST", "/memory/recall", { query, limit });
    }

    async timeline() {
        return this.client.request("GET", "/memory/timeline");
    }

    async insights() {
        return this.client.request("GET", "/memory/insights");
    }

    async profile() {
        return this.client.request("GET", "/memory/profile");
    }
}

File: sdk/python/memoryos/memory.py
from .client import MemoryClient
from typing import Dict, Any, List

class Memory:
    def __init__(self, api_key: str, base_url: str = "http://localhost:8000"):
        self.client = MemoryClient(api_key, base_url)

    def store(self, content: str, metadata: Dict = None) -> Dict[str, Any]:
        """Stores a new memory event."""
        return self.client.request("POST", "/memory/store", data={"content": content, "metadata": metadata})

    def recall(self, query: str, limit: int = 10) -> Dict[str, Any]:
        """Retrieves relevant memories based on a query."""
        return self.client.request("POST", "/memory/recall", data={"query": query, "limit": limit})

    def timeline(self) -> Dict[str, Any]:
        """Retrieves the chronological event timeline."""
        return self.client.request("GET", "/memory/timeline")

    def insights(self) -> Dict[str, Any]:
        """Retrieves reflection-based insights."""
        return self.client.request("GET", "/memory/insights")

    def profile(self) -> Dict[str, Any]:
        """Retrieves the user's cognitive profile summary."""
        return self.client.request("GET", "/memory/profile")
